cds_ils/migrator/utils.py

```diff
@@ -35,132 +42,77 @@ def pick(obj, *keys):
-                record["circulation_restriction"] = options[
-                    record["loan_period"]
-                ]
-            except KeyError:
-                raise ItemMigrationError(
-                    "Unknown circulation restriction (loan period) on "
-                    "barcode {0}: {1}".format(
-                        record["barcode"], record["loan_period"]
-                    )
-                )
-        del record["loan_period"]
-
-
-def clean_item_status(record):
-    """Translates item statuses."""
-    # possible values:
-    #   on shelf, missing, on loan, in binding, on order,
-    #   out of print, not published, not arrived, under review
-    options = {
-        "on shelf": "CAN_CIRCULATE",
-        "missing": "MISSING",
-        "on loan": "CAN_CIRCULATE",
-        "in binding": "IN_BINDING",
-        "FOR_REFERENCE_ONLY": "CAN_CIRCULATE",
-    }
+@contextmanager
+def commit():
+    """Commit transaction or rollback in case of an exception."""
     try:
-        record["status"] = options[record["status"]]
-    except KeyError:
-        raise ItemMigrationError(
-            "Unknown item status {0} on barcode {1}".format(
-                record["status"], record["barcode"]
-            )
-        )
-
-
-def clean_description_field(record):
-    """Cleans the item description."""
-    if record["description"] == "-" or record["description"] is None:
-        del record["description"]
-
-
-def clean_item_record(record):
-    """Clean the item record object."""
-    clean_circulation_restriction(record)
-    clean_item_status(record)
-    clean_description_field(record)
-    record["shelf"] = record["location"]
-    record["medium"] = "PAPER"  # requested as default value
-    record["created_by"] = {"type": "script", "value": "migration"}
-    del record["location"]
-    del record["id_bibrec"]
-    del record["id_crcLIBRARY"]
+        yield
+        db.session.commit()
+    except Exception:
+        print("Rolling back changes...")
+        db.session.rollback()
+        raise
+
+
+def reindex_pidtype(pid_type):
+    """Reindex records with the specified pid_type."""
+    click.echo('Indexing pid type "{}"...'.format(pid_type))
+    cli = create_cli()
+    runner = current_app.test_cli_runner()
+    runner.invoke(
+        cli,
+        "index reindex --pid-type {} --yes-i-know".format(pid_type),
+        catch_exceptions=True,
+    )
+    runner.invoke(cli, "index run", catch_exceptions=True)
+    click.echo("Indexing completed!")
+
+
+def bulk_index_records(records):
+    """Bulk index a list of records."""
+    indexer = RecordIndexer()
+
+    click.echo("Bulk indexing {} records...".format(len(records)))
+    indexer.bulk_index([str(r.id) for r in records])
+    indexer.process_bulk_queue()
+    click.echo("Indexing completed!")
+
+
+def model_provider_by_rectype(rectype):
+    """Return the correct model and PID provider based on the rectype."""
+    if rectype in ("serial", "multipart"):
```

Comment:
Shouldn't `journal` be added here?

Reply:
for the time being it is not used for journals, but I will add it for consistency